### DISCUSSION TITLE
Setting Room Size=0 In Reverb Still Produces Reverb

### DISCUSSION THREAD

**Discussion by yosimba2000** [godotengine/godot-proposals#11125] [Engine Core]
So this is focusing mainly on the Reverb values, but this can be applied to other values in other audio effects as well.

The `Room Size` slider goes from 0 to 1 in steps of 0.01, meaning it's relative. Relative to what, though?
I'm just ballparking this based on comparisons with Youtube videos, but when `Room Size` == 1, it seems to emulate a cathedral.
I estimate a cathedral with this reverb is about 100,000m<sup>3</sup>, or 46.5m x 46.5m x 46.5m.

But, the smallest VALID value you can assign to `Room Value` is 0.01 (assigning 0 should create absolutely no reverb, but it is currently not the case #11126). This means the smallest room that Reverb can simulate is 0.01 * 100,000m<sup>3</sup>, or 1000m<sup>3</sup>, which is a 10m x10m x10m room.

A 1000m<sup>3</sup> room is pretty large, and basically excludes designing reverb for smaller rooms. AKA you can't do reverb for bathrooms, which can be pretty reverberant even if they are small.

The value type of `Room Size` is already a float, so I don't see why Godot should automatically round values less than 0.01 directly to 0.
Let's say we want to simulate reverb for a room of 1m<sup>3</sup>, or 1m x 1m x 1m. This requires `Room Size` to be set to 0.00001 (if 1 equates to 100,000m<sup>3</sup>), which should be easily supported by floats.

Here are pictures to show 100,000m<sup>3</sup> and 1000m<sup>3</sup> rooms with human for scale.

![1000](https://github.com/user-attachments/assets/7cedcdfd-81e3-4929-ae42-bf51770df7af)
![100000](https://github.com/user-attachments/assets/e9bf2cb0-7622-481e-8b95-cf9a2ce9cb31)

**Comment by sockeye-d** [comment]
You can *normally* adjust your float slider precision steps with the `interface/inspector/default_float_step` setting, so it's weird that that effect specifically overrides it

**Reply by yosimba2000** [reply]
I thought this too. My editor settings have float step set to 0.00001.

Now, I can set the `room size` to something smaller than 0.01 via code, and returning the value does confirm it's not rounding it back to 0, but at the same time, the reverb effect treats all values less than 0.01 the same. That is, `room size` = 0.009 has the same amount of reverb as `room size` = 0.001 as `room size` = 0.0

So I guess it is 2 issues.
1) Changing reverb's room size via editor limits to 2 decimal places, regardless of editor's `default float step` setting.
2) All room sizes below 0.01 are treated the same. Also, room size = 0 still produces reverb 
#11126

**Comment by Mickeon** [comment]
The code of interest is here:
https://github.com/godotengine/godot/blob/0f5f3bc9546b46b2029fc8896dc859697f1eab97/servers/audio/effects/reverb_filter.cpp#L277-L296

I'm not much of an audio guy, but the feedback here looks to be clamped between `0.7` and `0.98`. So if the room size is a value very close to `0.0`, the feedback will still be a bit greater than `0.7`, which may have something to do to justify this.

**Reply by yosimba2000** [reply]
Possibly! There is also a second `feedback` parameter under the Predelay section in the Reverb effect (no idea if it's the same one shown in the code above).

**Comment by yosimba2000** [comment]
It worked! As per Mickeon's investigation, I made the following changes in `reverb_filter.cpp`
https://github.com/godotengine/godot/blob/0f5f3bc9546b46b2029fc8896dc859697f1eab97/servers/audio/effects/reverb_filter.cpp#L277-L296

Commented out lines 285-289. Changed line 284 to `c.feedback = params.room_size;`

Reverb Room size is longer clamped between `room_offset` and `room_offset + room_scale`.
Setting Room Size to 0 now produces 0 reverb, even when the Wet effect is turned up to 1!

I have no idea how to submit a fix, but just FYI in case someone think it's a good change.
